specification/azurestack/resource-manager/Microsoft.AzureStack/stable/2017-06-01/Product.json

```diff
@@ -150,6 +150,54 @@
         }
       }
     },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroup}/providers/Microsoft.AzureStack/registrations/{registrationName}/products/{productName}/listProducts": {
```

Comment:
>listProducts [](http://example.com/codeflow?start=155&length=12)

why do you need a POST .. this should really be modelled as a GET collection at the top level i.e : 

GET : /subscriptions/{subscriptionId}/resourceGroups/{resourceGroup}/providers/Microsoft.AzureStack/registrations/{registrationName}/products

Reply:
This PR is for fixing a swagger completeness item in s360. This API has been around for the entire lifetime of the RP, but wasn't documented for some reason. Changing how we model the operation would be a breaking change, and also wouldn't fix the swagger completeness issue. I don't know why this didn't show up earlier in swagger completeness - logs show the operation has been in use for a long time.

This action is modeled as a POST request because it has a body, the DeviceConfigurationParameter in the swagger JSON.

Reply:
Further, this action is identical to the POST ...{productName}/getProducts action. I don't know why they both exist - anyone who wrote these APIs is long gone.

Reply:
- I think the reason you have identical apis is that customers wanted to call this in ARM templates which can only be done for POST actions prefixed with "list*" as explained here : [https://docs.microsoft.com/en-us/azure/azure-resource-manager/templates/template-functions-resource#list](https://docs.microsoft.com/en-us/azure/azure-resource-manager/templates/template-functions-resource#list) . You probably should start planning to deprecate the getproducts one.

- Can you please annotate the  "DeviceConfigurationParameter" as coming from request body so that it is clear

You would need to work with the breaking change board to get the "NewApiversionRequired" Tag removed. Once that is done then please remove the "ARMChangesRequested" label and we can do the arm sign off

Reply:
Thank you. Can you give an example of what we should do to annotate the parameter? If you look at the parameter definition further down the file it does say "in": "body". Not sure what else we should do.